MoviesGo/MoviesGo/MenuModel/ViewModel/MenuViewModel.swift

```diff
@@ -0,0 +1,38 @@
+// MenuViewModel.swift
+// Copyright © Boris. All rights reserved.
+
+import Foundation
+
+protocol MenuViewModelProtocol: AnyObject {
+    var pageDataMovie: PageDataMovie? { get set }
+    func loadData(_ completion: @escaping () -> (Void))
+    var networkLayer: NetWorkLayer? { get set }
+}
+
+final class MenuViewModel: MenuViewModelProtocol {
+    // MARK: - Internal properties
+
+    var networkLayer: NetWorkLayer?
+    var pageDataMovie: PageDataMovie?
+
+    // MARK: - Internal function
+
+    func loadData(_ completion: @escaping () -> (Void)) {
+        networkLayer?.fetchData(PageDataMovie.self, methodStr: "movie/popular") { [weak self] result in
+            guard let self = self else { return }
+            switch result {
+            case let .failure(error):
+                switch error {
+                case let .failure(error):
```

Comment:
```suggestion
                case let .failure(error):
```
пропсы во вью юзают а не тут глянь у Жени с котом